### PR TITLE
[WIP] Add nodelocaldns image and template

### PR DESCRIPTION
**Pull request by superseb**
This is WIP until we decide in what k8s version we are adding nodelocal

https://github.com/rancher/rke/issues/1050

Requires: https://github.com/rancher/types/pull/980

### CODE REVIEW

File: rke/templates/templates.go [hunk 1]
```diff
@@ -86,6 +89,9 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			">=1.15.0-rancher0 <=1.15.3-rancher1-1": nginxIngressv18,
 			">=1.15.3-rancher2":                     nginxIngressV115,
 		},
+		Nodelocal: {
+			">=1.8.0-rancher0": nodelocalv116,
```
Comment (superseb) [inline rke/templates/templates.go]: @kinarashah Is this the expected behavior, I need to match this to all versions else it errors out while this template will only be introduced for v1.16.

Comment (kinarashah) [reply rke/templates/templates.go]: @superseb I think we'd have to add some exception in validation check for this template

Comment (kinarashah) [reply rke/templates/templates.go]: Opened `https://github.com/rancher/kontainer-driver-metadata/pull/76` to relax the check

**Comment by alena1108** [comment]
@superseb we need to introduce a new k8s version(s) for every major version supported by 2.4.x. And only expose this image in these versions, if possible

**Comment by superseb** [comment]
Superseded by https://github.com/rancher/kontainer-driver-metadata/pull/149

**Pull request closed without merging**